### PR TITLE
Backport: Fix alertmanager per tenant override config behavior (#7805)

### PR DESCRIPTION
**Pull request by dannykopping**
Backport #7805 into 2.7.x

### CODE REVIEW

File: pkg/logql/syntax/parser.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func validateSampleExpr(expr SampleExpr) error {
 		}
 
 		return validateSampleExpr(e.RHS)
-	case *LiteralExpr:
+	case *LiteralExpr, *VectorExpr:
```
Comment (dannykopping) [inline pkg/logql/syntax/parser.go:174]: This is from #6936

**Review by MasslessParticle — APPROVED** [review]
Followed the Review this morning. LGTM

**Comment by dannykopping** [comment]
@aminesnow backported, thank you :+1:

**Comment by aminesnow** [comment]
> @aminesnow backported, thank you +1

Thank you!

**Pull request merged**